specs/lightclient/beacon-chain.md

```diff
@@ -0,0 +1,186 @@
+    i, sync_committee_indices = 0, []
+    while len(sync_committee_indices) < SYNC_COMMITTEE_SIZE:
+        shuffled_index = compute_shuffled_index(uint64(i % active_validator_count), active_validator_count, seed)
+        candidate_index = active_validator_indices[shuffled_index]
+        random_byte = hash(seed + uint_to_bytes(uint64(i // 32)))[i % 32]
+        effective_balance = state.validators[candidate_index].effective_balance
+        if effective_balance * MAX_RANDOM_BYTE >= MAX_EFFECTIVE_BALANCE * random_byte:
+            sync_committee_indices.append(candidate_index)
+        i += 1
+    return sync_committee_indices
+```
+
+#### `get_sync_committee`
+
+```python
+def get_sync_committee(state: BeaconState, epoch: Epoch) -> SyncCommittee:
+    """
+    Return the sync committee for a given state and epoch.
+    """
+    indices = get_sync_committee_indices(state, epoch)
+    validators = [state.validators[index] for index in indices]
+    pubkeys = [validator.pubkey for validator in validators]
+    aggregates = [
+        bls.AggregatePKs(pubkeys[i:i + SYNC_COMMITTEE_PUBKEY_AGGREGATES_SIZE])
+        for i in range(0, len(pubkeys), SYNC_COMMITTEE_PUBKEY_AGGREGATES_SIZE)
+    ]
+    return SyncCommittee(pubkeys, aggregates)
+```
+
+### Block processing
+
+```python
+def process_block(state: BeaconState, block: BeaconBlock) -> None:
+    phase0.process_block(state, block)
+    process_sync_committee(state, block.body)
+```
+
+#### Sync committee processing
+
+```python
+def process_sync_committee(state: BeaconState, body: BeaconBlockBody) -> None:
+    # Verify sync committee aggregate signature signing over the previous slot block root
+    previous_slot = max(state.slot, Slot(1)) - Slot(1)
+    committee_indices = get_sync_committee_indices(state, get_current_epoch(state))
+    participant_indices = [committee_indices[i] for i in range(len(committee_indices)) if body.sync_committee_bits[i]]
+    participant_pubkeys = [state.validators[participant_index].pubkey for participant_index in participant_indices]
+    domain = get_domain(state, DOMAIN_SYNC_COMMITTEE, compute_epoch_at_slot(previous_slot))
+    signing_root = compute_signing_root(get_block_root_at_slot(state, previous_slot), domain)
+    assert bls.FastAggregateVerify(participant_pubkeys, signing_root, body.sync_committee_signature)
+
+    # Reward sync committee participants
+    participant_rewards = Gwei(0)
+    active_validator_count = uint64(len(get_active_validator_indices(state, get_current_epoch(state))))
+    for participant_index in participant_indices:
+        base_reward = get_base_reward(state, participant_index)
+        reward = Gwei(base_reward * active_validator_count // len(committee_indices) // SLOTS_PER_EPOCH)
+        increase_balance(state, participant_index, reward)
+        participant_rewards += reward
+
+    # Reward beacon proposer
+    increase_balance(state, get_beacon_proposer_index(state), Gwei(participant_rewards // PROPOSER_REWARD_QUOTIENT))
```

Comment:
Should this new sync committee reward component be deducted from `get_inactivity_penalty_deltas`((https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/beacon-chain.md#components-of-attestation-deltas) `penalties` component for the well-performed validators? i.e., modify line `penalties[index] += Gwei(BASE_REWARDS_PER_EPOCH * base_reward - get_proposer_reward(state, index))`.

Reply:
^^^^ The above question was answered during the call, do not need to deduct the sync committee rewards.

Another question about light client reward + inactivity leak accounting:

- Since other reward components are all accounted during **epoch-processing**, they were calculated with (1) if `is_in_inactivity_leak(state)` is True, then pre-issue the `base_reward` and (2) cancel the pre-issued reward in `get_inactivity_penalty_deltas`.
- However, the light client committee rewards are calculated during the **block-processing**. So it cannot be detected with `is_in_inactivity_leak(state)`.
- It seems #2140 form can fix this issue well by using flags. Does `lightclient` patch require `accounting-reform` patch? Or there is a better way to delay the calculation for this case?

Reply:
Update: applied @djrtwo 's suggestion in #2141 (https://github.com/ethereum/eth2.0-specs/pull/2141/commits/1b7b842aaff3c133266f67b2c0dd6811e580bf88)